File: ruby/client_sync.rb
require 'celluloid/io'
require 'http'

class NapFetcher
  
  def request_nap(msg)
    HTTP.get(
      "http://localhost:4567?sec=3&msg=#{msg}", 
      socket_class: Celluloid::IO::TCPSocket
    )
  end
  
end

fetcher = NapFetcher.new

words = %w(i am fast)

words.each do |word|
  response = fetcher.request_nap(word)
  puts response.body
end

File: python/client_sync.py
import tornado.httpclient

http_client = tornado.httpclient.HTTPClient()

words = ['i', 'am', 'fast']

for word in words:
  response = http_client.fetch("http://localhost:4567?sec=3&msg=%s" % (word))
  if response.error:
      print("Error:", response.error)
  else:
      print(response.body)
